Description: growpart: fix bug that stopped GPT disks from being grown past 2TB.
 MBR max size was being applied to GPT partitioned disks.
 The change here is to only apply the mbr max (and only WARN about the limit)
 if the format is MBR.
 . 
 Also, adjust test/test-growpart to support reading disk orig size
 and new size from environment to make testing more easy.
 .
   sudo PATH=./bin:$PATH PT_TYPE=gpt \
      DISK_SIZE_NEW=5T DISK_SIZE_ORIG=1T ./test/test-growpart
 .
 This is a combination of bzr revisions 323 and 324.
 .
 The changes to test/test-growpart have been dropped as they did not
 apply cleanly.
Bug: https://launchpad.net/bugs/1762748
Origin: upstream, bzr-revno-323 and bzr-revno-324
Applied-upstream: commit: revno 324

=== modified file 'bin/growpart'
--- a/bin/growpart
+++ b/bin/growpart
@@ -237,8 +237,6 @@ resize_sfdisk() {
 	fi
 
 	debug 1 "$sector_num sectors of $sector_size. total size=${disk_size} bytes"
-	[ $(($disk_size/512)) -gt $mbr_max_512 ] &&
-		debug 1 "WARN: disk is larger than 2TB. additional space will go unused."
 
 	rqe sfd_dump sfdisk --unit=S --dump "${DISK}" >"${dump_out}" ||
 		fail "failed to dump sfdisk info for ${DISK}"
@@ -282,17 +280,21 @@ resize_sfdisk() {
 		[ -n "${max_end}" ] ||
 		fail "failed to get max_end for partition ${PART}"
 
-	mbr_max_sectors=$((mbr_max_512*$((sector_size/512))))
-	if [ "$max_end" -gt "$mbr_max_sectors" ]; then
-		max_end=$mbr_max_sectors
-	fi
-
 	if [ "$format" = "gpt" ]; then
 		# sfdisk respects 'last-lba' in input, and complains about
 		# partitions that go past that.  without it, it does the right thing.
 		sed -i '/^last-lba:/d' "$dump_out" ||
 			fail "failed to remove last-lba from output"
 	fi
+	if [ "$format" = "dos" ]; then
+		mbr_max_sectors=$((mbr_max_512*$((sector_size/512))))
+		if [ "$max_end" -gt "$mbr_max_sectors" ]; then
+			max_end=$mbr_max_sectors
+		fi
+		[ $(($disk_size/512)) -gt $mbr_max_512 ] &&
+			debug 0 "WARNING: MBR/dos partitioned disk is larger than 2TB." \
+				"Additional space will go unused."
+	fi
 
 	local gpt_second_size="33"
 	if [ "${max_end}" -gt "$((${sector_num}-${gpt_second_size}))" ]; then
